### PR TITLE
Improve santa tracker build comparisons for vfs retention/instant execution

### PR DESCRIPTION
**Pull request by wolfs**
This PR cleans up `FasterIncrementalAndroidBuildsPerformanceTest` and makes it more in line with https://github.com/gradle/santa-tracker-performance.

### CODE REVIEW

File: buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/dependencies/DependenciesMetadataRulesPlugin.kt [hunk 1]
```diff
@@ -78,6 +78,10 @@ open class DependenciesMetadataRulesPlugin : Plugin<Project> {
                 // Test dependencies - minify: remove unused transitive dependencies
                 withLibraryDependencies("org.gradle.org.littleshoot:littleproxy", DependencyRemovalByNameRule::class,
                     setOf("barchart-udt-bundle", "guava", "commons-cli"))
+
+                // Gradle profiler should use the bundled tooling API
+                withLibraryDependencies("org.gradle.profiler:gradle-profiler", DependencyRemovalByNameRule::class,
+                    setOf("gradle-tooling-api"))
```
Comment (lptr) [inline buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/dependencies/DependenciesMetadataRulesPlugin.kt]: Should we break up `gradle-profiler` in a way that we could depend on a tooling API-less variant/module?

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/FasterIncrementalAndroidBuildsPerformanceTest.groovy [hunk 2]
```diff
@@ -131,4 +135,24 @@ class FasterIncrementalAndroidBuildsPerformanceTest extends AbstractCrossBuildPe
...
+                    GradleEnterprisePluginSettingsFixture.applyEnterprisePlugin(settingsFile)
+                }
+
+                @Override
+                void afterScenario(ScenarioContext context) {
+                    settingsFile.text = originalSettingsFileText
+                }
+            }
+        }
+    }
```
Comment (lptr) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/FasterIncrementalAndroidBuildsPerformanceTest.groovy]: Would it make sense to add this to `gradle-profiler`?

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/FasterIncrementalAndroidBuildsPerformanceTest.groovy [hunk 3]
```diff
@@ -111,6 +111,7 @@ class FasterIncrementalAndroidBuildsPerformanceTest extends AbstractCrossBuildPe
             builder.invocation.args(AndroidGradlePluginVersions.OVERRIDE_VERSION_CHECK)
             builder.invocation.args("-Dorg.gradle.workers.max=8", "--no-build-cache", "--no-scan")
             builder.invocation.useToolingApi()
+            builder.invocationCount(100)
```
Comment (lptr) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/FasterIncrementalAndroidBuildsPerformanceTest.groovy]: I guess we could roll this down to something like 60. WDYT?

BTW, are we down to 1 warmups here, too?

File: buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/dependencies/DependenciesMetadataRulesPlugin.kt (hunk 1 above)
Comment (wolfs) [reply buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/dependencies/DependenciesMetadataRulesPlugin.kt]: I don't think so. The right solution is that our project dependency should conflict with the external dependency and then the right dependency is selected. @jjohannes wanted to look into why this is not the case currently.

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/FasterIncrementalAndroidBuildsPerformanceTest.groovy (hunk 2 above)
Comment (wolfs) [reply subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/FasterIncrementalAndroidBuildsPerformanceTest.groovy]: Yeah, probably via an init script `gradle-profiler` can already generate.

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/FasterIncrementalAndroidBuildsPerformanceTest.groovy (hunk 3 above)
Comment (wolfs) [reply subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/FasterIncrementalAndroidBuildsPerformanceTest.groovy]: I now use 1 warmup and 60 iterations.

File: buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/dependencies/DependenciesMetadataRulesPlugin.kt (hunk 1 above)
Comment (jjohannes) [reply buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/dependencies/DependenciesMetadataRulesPlugin.kt]: This is the problem: https://github.com/gradle/gradle/issues/12002

I think this workaround here is ok for now. Maybe you can clarify this in the comment and add a link to the GH issue.

Comment (wolfs) [reply buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/dependencies/DependenciesMetadataRulesPlugin.kt]: I added a TODO.

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/FasterIncrementalAndroidBuildsPerformanceTest.groovy (hunk 2 above)
Comment (lptr) [reply subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/FasterIncrementalAndroidBuildsPerformanceTest.groovy]: Are you planning to address this in this PR or shall we create an issue?

Comment (wolfs) [reply subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/FasterIncrementalAndroidBuildsPerformanceTest.groovy]: I am not. https://github.com/gradle/gradle-profiler/issues/191

**Pull request merged**